### PR TITLE
Mry/email

### PR DESCRIPTION
**Pull request by mry-usds**
This PR does ...

## Changes
- Adds support for multiple transports
- Adds support for email transport (link is not working)
- Adds support for prime-data-hub web page

## Checklist
- [X] Tested locally?
- [X] Added new dependencies?
- [X] Updated the docs?
- [ ] Added a test?
- [X] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Fixes 

- #issue

## To Be Done

- Need to retrieve the file by id

### CODE REVIEW

File: prime-router/src/main/kotlin/transport/EmailTransport.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package gov.cdc.prime.router.transport
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/transport/EmailTransport.kt:1]: Hmm.  Technically, this isn't really a 'Transport' since nothing's getting transported.    The transport mechanism is like "make the file available for download" or something like that? 

And sending an email is a generic activity - can we genericize this code, make it static, move out the 'from' and 'subject' (which I'm guessing you were intending to do anyways) and just have a general emailing facility that can contact anybody about anything?

**Review by jimduff-usds — COMMENTED** [review]
See comments inline.

File: prime-router/pom.xml [hunk 2]
```diff
@@ -139,6 +139,8 @@
                                         <include>metadata/**/*.schema</include>
                                         <include>metadata/**/*.valuesets</include>
                                         <include>metadata/**/*.csv</include>
+                                        <include>assets/**/*__inline.html</include>
+                                        <include>metadata/pdi-covid-19.csv</include>
```
Comment (RickHawesUSDS) [inline prime-router/pom.xml:143]: Look's like a mistake?

File: prime-router/src/main/kotlin/OrganizationService.kt [hunk 3]
```diff
@@ -94,21 +96,28 @@ data class OrganizationService(
...
-        val filePath: String = "."
-    ) {
-        enum class TransportType {
-            SFTP,
-            DEFAULT
-            // EMAIL
-            // DROPBOX
-            // API
-        }
+    enum class BatchTimeZone(val zoneId: String) {
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/OrganizationService.kt:99]: I've moved this to separate class.

File: prime-router/src/main/kotlin/azure/DownloadFunction.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        if( file.isBlank() ) 
+            return responsePage(request);
+        else 
+            return responseFile(request,file);
+    }
+
+    private fun responseFile( request: HttpRequestMessage<String?>, fileName: String ) : HttpResponseMessage {
+
+        val body = Files.readString(Path.of("./metadata/pdi-covid-19.csv"))
+
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/azure/DownloadFunction.kt:53]: Please don't mix results and metadata.

Comment (RickHawesUSDS) [reply prime-router/src/main/kotlin/azure/DownloadFunction.kt:53]: Why is this hard coded?

File: prime-router/src/main/kotlin/azure/DownloadFunction.kt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        val previousTestResults = listOf<TestResult>(
+            TestResult( GregorianCalendar(2020,11,10), 7, 8, 2, "123111111"),
+            TestResult( GregorianCalendar(2020,11,9), 6, 11, 2, "123111111"),
+            TestResult( GregorianCalendar(2020,11,8), 5, 12, 3, "123111111"),
+            TestResult( GregorianCalendar(2020,11,7), 4, 9, 1, "123111111"),
+            TestResult( GregorianCalendar(2020,11,6), 3, 9, 1, "123111111"),
+            TestResult( GregorianCalendar(2020,11,5), 2, 10, 3, "123111111"),
+            TestResult( GregorianCalendar(2020,11,4), 1, 12, 4, "123111111")            
+        )
+
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/azure/DownloadFunction.kt:79]: Not sure what this is.

File: prime-router/src/main/kotlin/azure/SendFunction.kt [hunk 6]
```diff
@@ -35,24 +36,19 @@ class SendFunction {
...
-                        )
-                        val content = workflowEngine.readBody(header)
-                        // TODO:  look up the correct class to call based on the transport metadata
-                        val transport = SftpTransport()
-                        transport.send(service, header, content)
+                val content = workflowEngine.readBody(header)
+                service.transports.forEach {
+                    when (it) {
+                        is OrganizationService.SFTP -> SftpTransport().send(service.fullName, it, header, content)
+                        is OrganizationService.Email -> EmailTransport().send(service.fullName, it, header, content)
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/azure/SendFunction.kt:43]: How do we record errors in one transport and not the other?

File: prime-router/src/test/csv_test_files/input/example-input-from-simplereport.csv [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
...atient_county,Patient_state,Patient_zip_code,Patient_phone_number,Patient_ID,Patient_lookup_ID,Employed_in_healthcare,Resident_congregate_setting,Testing_lab_specimen_ID,Test_result_coded,Specimen_collection_date_time,Ordering_provider_ID,First_test,Symptomatic_for_disease,Testing_lab_name,Testing_lab_CLIA,Testing_lab_city,Testing_lab_state,Testing_lab_street,Testing_lab_street2,Testing_lab_zip_code,Testing_lab_phone_number,Ordering_facility_city,Odering_facility_county,Ordering_facility_name,Ordering_facility_phone_number,Ordering_facility_state,Ordering_facility_street,Ordering_facility_street2,Ordering_facility_zip_code,Ordering_provider_last_name,Ordering_provider_first_name,Ordering_provider_street,Ordering_provider_zip_code,Ordering_provider_phone,Ordered_test_code,Specimen_source_site_code,Specimen_type_code,Device_ID,Instrument_ID,Test_date,Date_result_released, Illness_onset_date, Order_test_date, Patient_street_2, Patient_role, Testing_lab_street_2, Testing_lab_county, Organization_name, Ordering_facility_street_2, Ordering_facility_county, Ordering_provider_street_2, Ordering_provider_city, Ordering_provider_state, Ordering_provider_phone_number, Ordering_provider_county
```
Comment (RickHawesUSDS) [inline prime-router/src/test/csv_test_files/input/example-input-from-simplereport.csv:1]: I'm confused why this change is needed.

**Review by RickHawesUSDS — CHANGES_REQUESTED** [review]
I suggest that you call a meeting about what you are trying to do.

**Comment by mry-usds** [comment]
Dropping in favor of some smaller requests
